core/src/main/scala/org/apache/spark/deploy/PythonRunner.scala

```diff
@@ -96,11 +96,13 @@ object PythonRunner {
     try {
       val process = builder.start()
 
-      new RedirectThread(process.getInputStream, System.out, "redirect output").start()
+      val redirectOutput = new RedirectThreadAndCatchErrorMsg(
+        process.getInputStream, System.out, "redirect output", "Traceback")
```

Comment:
I think we should
1. remove `builder.redirectErrorStream`
2. find an easy way to merge both stderr & stdout to print out later
3. extract stderr from `process`
4. use it in exception message

Reply:
> I think we should
> 
> 1. remove `builder.redirectErrorStream`
> 2. find an easy way to merge both stderr & stdout to print out later (manually on JVM)
> 3. extract stderr from `process`
> 4. use it in exception message

You mean the whole stderr as exception message?

Reply:
> I think we should
> 
> 1. remove `builder.redirectErrorStream`
> 2. find an easy way to merge both stderr & stdout to print out later (manually on JVM)
> 3. extract stderr from `process`
> 4. use it in exception message

Since traceback is print before Process exit, so we still need to extract error message from output. Or we just use all information in stderr as exception message?

Reply:
gentle ping @HyukjinKwon

Reply:
yeah, i thought to show the stderr as error message. I don't think there's a claver way to extract Python exception cleanly.

Reply:
> yeah, i thought to show the stderr as error message. I don't think there's a claver way to extract Python exception cleanly.

In current code can satisfy this requirement, but can't keep  stdout and stderr all sync.